src/main/java/strategies/SearchStrategy.java

```diff
@@ -0,0 +1,14 @@
+package strategies;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public interface SearchStrategy {
+
+    List<String> search(List<String> list, Map<String, ArrayList<Integer>> searchMap, String query);
+
+    default String[] queryWords(String query) {
```

Comment:
You are using the same operation in LineTokenizer. Can you create the utility class with a meaningful name in the `util` package and put this method there?
See an example of Utility class - https://rules.sonarsource.com/java/RSPEC-1118

Reply:
created new class